clear-intermediate-datasets/python-runnables/clear_int_datasets/runnable.py

```diff
@@ -77,8 +77,13 @@ def run(self, progress_callback):
         logging.info("Found {} FLOW OUTPUT datasets: {}".format(str(len(flow_outputs)),
                                                                 str(flow_outputs)))
 
+        #Identify Standaone datasets (not connected to another object)
+        standalone_datasets = [dataset["name"] for dataset in all_datasets if dataset["name"] not in input_datasets + output_datasets]
+        logging.info("Found {} STANDALONE datasets: {}".format(str(len(standalone_datasets)),
+                                                                str(standalone_datasets)))
+        
         # Identify Intermediate datasets:
-        intermediate_datasets = [dataset["name"] for dataset in all_datasets if dataset["name"] not in flow_inputs + flow_outputs]
+        intermediate_datasets = [dataset["name"] for dataset in all_datasets if dataset["name"] not in flow_inputs + flow_outputs + standalone_datasets]
```

Comment:
looping several times on all_datasets (for standalones, intermediate, partitonned). It could be very long when using the macro on big flows. A unique loop to create the different list would be more efficient

Reply:
Done in commit [dde48c0f803f4c111cc4be6b24c6051d69906e0](https://github.com/gbetegon88/dataiku-contrib/commit/dde48c0f803f4c111cc4be6b24c6051d69906e02#diff-6f25a7eaa37b1d34604450d8bc9334c0558917e69a3eb97e69bea9d70378e561). Standalone, intermediate and partitioned datasets are now tagged in just one loop.